common/lockless_queue.c

```diff
@@ -0,0 +1,136 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include <openenclave/internal/lockless_queue.h>
+#include <stdlib.h>
+
+/* functions for lockless_queue_node */
+/*---------------------------------------------------------------------------*/
+void init_lockless_queue_node(lockless_queue_node* p_node)
```

Comment:
To avoid symbol collistions, the naming convention is to use oe_ prefix for all non static functoins.